ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/Cli.scala

```diff
@@ -169,6 +169,11 @@ object Cli {
            |Party names must be their hints.""".stripMargin)
 
     opt[Unit]("list")
+      .action((_, c) => c.copy(listTestSuites = true))
+      .text(
+        """Lists all available test suites that can be used in the include and exclude options. Test names always start with their suite name, so using the suite name as a prefix matches all tests in a given suite.""")
+
+    opt[Unit]("list-all")
```

Comment:
What happens if you use both `--list` and `--list-all`?

Reply:
At the moment `--list` wins over everything. So if you say `--include=... --perf-tests=... --list` you still get the list of test suites. That's very much related to @cocreature's comment: we have a bunch of "flat" flags (i.e. no hierarchy) as far as the external interface is concerned, but internally we treat them as special things where some affect others. I think the tool has become rich enough that it's now worth having explicit subcommands, each which its own set of allowed flags.